app/controllers/provider_foreman_controller.rb

```diff
@@ -189,6 +189,18 @@ def provider_active_tree?
 
   private
 
+  def provider_foreman_pause
+    @record = find_record(ExtManagementSystem, params[:id])
+    @record.pause! # TODO: reload toolbar button enablement after the pause
```

Comment:
@martinpovolny I need your help about this, if it is even possible and if not, how else should I do this?

Reply:
You need to queue it. There should be plenty of examples using `initiate_wait_for_task` ;).